scripts/deployKovan.ts

```diff
@@ -4,55 +4,50 @@
 require('dotenv').config()
 import { JsonRpcProvider } from '@ethersproject/providers'
 import hre from 'hardhat'
-import { assert } from 'ts-essentials'
+import { mapValues } from 'lodash'
 
 import { ZERO_GAS_OPTS } from '../test-e2e/helpers/utils'
-import { deploy } from './common'
+import { deploy, getRequiredEnv } from './common'
 
 // optimism's addresses: https://github.com/ethereum-optimism/optimism/tree/master/packages/contracts/deployments
 
-const L1_PAUSE_PROXY_ADDRESS = '0x0e4725db88Bb038bBa4C4723e91Ba183BE11eDf3'
-const L1_ESM_ADDRESS = '0xD5D728446275B0A12E4a4038527974b92353B4a9'
-const L1_DAI_ADDRESS = '0x4F96Fe3b7A6Cf9725f59d353F723c1bDb64CA6Aa'
-assert(process.env.KOVAN_DEPLOYER_PRIV_KEY, 'Please provide KOVAN_DEPLOYER_PRIV_KEY in .env file')
-const L1_DEPLOYER_PRIV_KEY = process.env.KOVAN_DEPLOYER_PRIV_KEY
-const L1_XDOMAIN_MESSENGER = '0x4361d0F75A0186C05f971c566dC6bEa5957483fD'
-const L1_RPC_URL = 'https://parity0.kovan.makerfoundation.com:8545'
-assert(process.env.L2_TESTNET_DEPLOYER_PRIV_KEY, 'Please provide L2_TESTNET_DEPLOYER_PRIV_KEY in .env file')
-const L2_DEPLOYER_PRIV_KEY = process.env.L2_TESTNET_DEPLOYER_PRIV_KEY
-const L2_RPC_URL = 'https://kovan.optimism.io/'
-const L2_XDOMAIN_MESSENGER = '0x4200000000000000000000000000000000000007'
+const L1_KOVAN_RPC_URL = getRequiredEnv('L1_KOVAN_RPC_URL')
+const L1_KOVAN_DEPLOYER_PRIV_KEY = getRequiredEnv('L1_KOVAN_DEPLOYER_PRIV_KEY')
+const L2_KOVAN_RPC_URL = getRequiredEnv('L2_KOVAN_RPC_URL')
+const L2_KOVAN_DEPLOYER_PRIV_KEY = getRequiredEnv('L2_KOVAN_DEPLOYER_PRIV_KEY')
+
+const L1_KOVAN_PAUSE_PROXY_ADDRESS = '0x0e4725db88Bb038bBa4C4723e91Ba183BE11eDf3'
+const L1_KOVAN_ESM_ADDRESS = '0xD5D728446275B0A12E4a4038527974b92353B4a9'
+const L1_KOVAN_DAI_ADDRESS = '0x4F96Fe3b7A6Cf9725f59d353F723c1bDb64CA6Aa'
+const L1_KOVAN_XDOMAIN_MESSENGER = '0x4361d0F75A0186C05f971c566dC6bEa5957483fD'
+const L2_KOVAN_XDOMAIN_MESSENGER = '0x4200000000000000000000000000000000000007'
```

Comment:
What is this address?

Reply:
This is predeployed on l2: 
https://github.com/ethereum-optimism/optimism/tree/master/packages/contracts/deployments